File: experiments/test_multiline_quotes.rs
use lino::parse_lino;

fn main() {
    println!("Testing multiline quotes as described in issue #53:");
    println!();

    let input = r#"(
  "long
string literal representing
the reference"

  'another
long string literal
as another reference'
)"#;

    println!("Input:");
    println!("{}", input);
    println!();

    match parse_lino(input) {
        Ok(result) => {
            println!("✓ Parsing succeeded!");
            println!("Parsed result: {:#?}", result);
        }
        Err(e) => {
            println!("✗ Parsing failed: {:?}", e);
        }
    }
}

File: experiments/test_multiline_quotes.cs
using System;
using Platform.Protocols.Lino;

static class Program
{
    static void Main()
    {
        var parser = new Parser();

        Console.WriteLine("Testing multiline quotes as described in issue #53:");
        Console.WriteLine();

        var input = @"(
  ""long
string literal representing
the reference""

  'another
long string literal
as another reference'
)";

        Console.WriteLine("Input:");
        Console.WriteLine(input);
        Console.WriteLine();

        try
        {
            var result = parser.Parse(input);
            Console.WriteLine("✓ Parsing succeeded!");
            Console.WriteLine($"Number of links: {result.Count}");

            foreach (var link in result)
            {
                Console.WriteLine($"Link: {link}");
            }
        }
        catch (Exception ex)
        {
            Console.WriteLine($"✗ Parsing failed: {ex.Message}");
            Console.WriteLine($"Exception type: {ex.GetType().Name}");
        }
    }
}